quinn-proto/src/frame.rs

```diff
@@ -414,10 +417,18 @@ impl FrameStruct for Stream {
     const SIZE_BOUND: usize = 1 + 8 + 8 + 8;
 }
 
-impl Stream {
+/// Metadata from a stream frame
```

Comment:
This specifically represents the metadata/header portion of a stream frame, rather than the state of a logical stream. Note that it's defined in the `frame` module and we only ever use qualified names to reference it, much like other frame types. I split it out so that `Streams::poll_transmit` does not have to construct a `Bytes`, which would require copying.

Reply:
Makes sense!